staging/src/k8s.io/cli-runtime/pkg/printers/omit_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type testResourcePrinter func(runtime.Object, io.Writer) error
+
+func (p testResourcePrinter) PrintObj(o runtime.Object, w io.Writer) error {
+	return p(o, w)
+}
+
+func checkManagedFields(object runtime.Object, w io.Writer) error {
```

Comment:
I would drop this function altogether and compare the exact thing that you want as the output. In other words, I would make `testResourcePrinter` take the expected object that you think it will receive, and test that it does. Maybe something like this:

```golang
type testResourcePrinter struct {
    expected runtime.Object
}

func (p testResourcePrinter) PrintObj(o runtime.Object, w io.Writer) error {
    if !reflect.DeepEqual(o, p.expected) {
        return fmt.Errorf("%v is different from expected: %v", o, p.expected)   
    }
    return nil
}
```

Then you testcase only needs the input and expected output. Does that make sense?

Reply:
Thanks, I think this is better.